### PR TITLE
Stable version to replace the shadow service in darwinia.js

### PR DESCRIPTION
**Pull request by clearloop**
+ [x] Adapt lowercase request method with slashes
+ [x] parse proof output
+ [x] support scale codec

### CODE REVIEW

File: rpc/server.go [hunk 1]
```diff
@@ -0,0 +1,602 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+// Copyright 2020 The Darwinia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```
Comment (clearloop) [inline rpc/server.go:6]: Modified the source code of `net/rpc` and `net/rpc/jsonrpc`, the LICENSE needs to be reviewed.

**Comment by clearloop** [comment]
## Version Mismatched

```
module github.com/darwinia-network/darwinia.go                                                                                                                                                              
                                                                                                                                                                                                            
go 1.14                                                                                                                                                                                                     
                                                                                                                                                                                                            
require (                                                                                                                                                                                                   
    github.com/ChainSafe/gossamer v0.0.0-20200426141239-54658ca603f6                                                                                                                                        
    github.com/elastic/gosigar v0.10.5 // indirect                                                                                                                                                          
    github.com/ethereum/go-ethereum v1.9.6                                                                                                                                                                  
    github.com/fsnotify/fsnotify v1.4.9 // indirect                                                                                                                                                         
    github.com/kr/text v0.2.0 // indirect                                                                                                                                                                   
    github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect                                                                                                                               
    github.com/olekukonko/tablewriter v0.0.4 // indirect                                                                                                                                                    
    github.com/spf13/cobra v1.0.0                                                                                                                                                                           
    github.com/steakknife/bloomfilter v0.0.0-20180922174646-6819c0d2a570 // indirect                                                                                                                        
    github.com/steakknife/hamming v0.0.0-20180906055917-c99c65617cd3 // indirect                                                                                                                            
    github.com/stretchr/testify v1.5.1 // indirect                                                                                                                                                          
    github.com/tranvictor/ethashproof v0.0.0-20200216154736-82a2b716eac4                                                                                                                                    
    golang.org/x/net v0.0.0-20200425230154-ff2c4b7c35a0 // indirect                                                                                                                                         
    golang.org/x/sys v0.0.0-20200420163511-1957bb5e6d1f // indirect                                                                                                                                         
    gopkg.in/check.v1 v1.0.0-20200227125254-8fa46927fb4f // indirect                                                                                                                                        
    gopkg.in/yaml.v2 v2.2.8 // indirect                                                                                                                                                                     
) 
```

The `ethashproof` has a copy of `ethash` at `v1.8.2`, just change few lines, it sucks, and `gossamer` requires the lastest `go-ethereum` at `v1.9.6`, this can make the `ethashproof` get broken, because go mod will auto-choose the newer version.

File: core/shadow.go [hunk 2]
```diff
@@ -31,27 +34,62 @@ func (s *Shadow) GetEthHeaderByNumber(
...
+			return err
+		}
+
+		fmt.Printf("%v\n", sheader)
+		fmt.Printf("%v\n", sproof)
+
+		*resp = GetEthHeaderWithProofByNumberCodecResp{
+			hex.EncodeToString(sheader),
+			hex.EncodeToString(sproof),
+		}
```
Comment (clearloop) [inline core/shadow.go:91]: Encode Bug

File: rpc/server.go (hunk 1 above)
Comment (HackFisher) [reply rpc/server.go:6]: BSD License should be good

**Pull request merged**